### ISSUE TITLE
Shinyproxy 3.0.1 and Rstudio can't upload Files

### ISSUE THREAD

**Issue by 0Xilef** [openanalytics/shinyproxy#419]
Hello,

first of all thanks for the great job with Shinyproxy.
I have a Problem and can't find out why.

I want to use Shinyproxy 3.0.1 or 3.0.0 with RStudio. But i can't upload any files with Version 3.0.1 or 3.0.0!
Size of files are small (1KB).
[test.txt](https://github.com/openanalytics/shinyproxy/files/11344445/test.txt)

If i use Shinyproxy 2.6.1 with RStudio i able to Upload files.

i tried following Browsers: Chrome and Fierfox

RStudio shows me the Error: "Unexpected response from server"
![shinyproxy_error](https://user-images.githubusercontent.com/70663840/234878680-9d2f5510-aa50-42e5-be1e-f67b68d1d7a0.PNG)

I use Shinyproxy with Docker Swarm

```
Linuxserver: Ubuntu 22.04
Kernel: 5.4.0-137-generic
Docker version: 23.0.3, build 3e7cbfd
```

## I can reproduce the error with this example:

**application.yml:**
```yml
proxy:
  port: 8080
  authentication: simple
  admin-groups: admins
  users:
  - name: jack
    password: password
    groups: admins
  - name: jeff
    password: password
  container-backend: docker-swarm
  docker:
    internal-networking: true
  specs:
    - id: rstudio
      container-image: openanalytics/shinyproxy-rstudio-ide-demo:2021.09.2_382__4.1.2
      container-network: sp-example-net
      container-env:
        DISABLE_AUTH: true
        # Use the following line when using ShinyProxy 2.6.0 or later
        WWW_ROOT_PATH: "#{proxy.getRuntimeValue('SHINYPROXY_PUBLIC_PATH')}"
        # Use the following line when using ShinyProxy 2.5.0
        # WWW_ROOT_PATH: "#{proxySpec.containerSpecs[0].env.get('SHINYPROXY_PUBLIC_PATH')}"
      port: 8787

logging:
  file: shinyproxy.log
```

**Dockerfile Shinyproxy:**

If i change to openanalytics/shinyproxy:2.6.1 then the File Upload is okay.

```dockerfile
FROM openanalytics/shinyproxy:3.0.1

COPY application.yml /opt/shinyproxy/application.yml
```

**Start the Service:**

```bash
docker build . -t shinyproxy-example

docker network create -d overlay --attachable sp-example-net

docker service create --name sp-test-service --mount type=bind,src=/var/run/docker.sock,dst=/var/run/docker.sock --publish 8080:8080 --network sp-example-net --group $(getent group docker | cut -d: -f3) shinyproxy-example
```

**Logs from Shinyproxy:**

```log
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.7.6)
2023-04-27 14:08:37.736  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode
2023-04-27 14:08:37.745  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Redis repositories in DEFAULT mode.
2023-04-27 14:08:37.834  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 64 ms. Found 0 Redis repository interfaces.
2023-04-27 14:08:38.063  INFO 1 --- [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'errorChannel' has been explicitly defined. Therefore, a default PublishSubscribeChannel will be created.
2023-04-27 14:08:38.095  INFO 1 --- [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'integrationHeaderChannelRegistry' has been explicitly defined. Therefore, a default DefaultHeaderChannelRegistry will be created.
2023-04-27 14:08:39.050  INFO 1 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.integration.config.IntegrationManagementConfiguration' of type [org.springframework.integration.config.IntegrationManagementConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-04-27 14:08:39.072  INFO 1 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'integrationChannelResolver' of type [org.springframework.integration.support.channel.BeanFactoryChannelResolver] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-04-27 14:08:39.979  WARN 1 --- [           main] io.undertow.websockets.jsr               : UT026010: Buffer pool was not set on WebSocketDeploymentInfo, the default pool will be used
2023-04-27 14:08:40.022  INFO 1 --- [           main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2023-04-27 14:08:40.022  INFO 1 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 5759 ms
2023-04-27 14:08:40.116  INFO 1 --- [           main] e.o.c.service.IdentifierService          : ShinyProxy runtimeId:                   6cd3e820-04c9-4cb5-8094-86379cc61828
2023-04-27 14:08:40.233  INFO 1 --- [           main] e.o.c.service.IdentifierService          : ShinyProxy instanceID (hash of config): d806838fc182a40e344636d9bc22837fd4538ba8
2023-04-27 14:08:44.356  INFO 1 --- [           main] o.s.boot.web.servlet.RegistrationBean    : Filter orderedFormContentFilter was not registered (disabled)
2023-04-27 14:08:44.921  WARN 1 --- [           main] org.thymeleaf.templatemode.TemplateMode  : [THYMELEAF][main] Template Mode 'HTML5' is deprecated. Using Template Mode 'HTML' instead.
2023-04-27 14:08:45.277  INFO 1 --- [           main] o.s.s.web.DefaultSecurityFilterChain     : Will secure any request with [org.springframework.security.web.session.DisableEncodeUrlFilter@2063c53e, org.springframework.security.web.session.ForceEagerSessionCreationFilter@2a9bc08f, org.springframework.security.web.session.ForceEagerSessionCreationFilter@e344ad3, org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@50f40653, org.springframework.security.web.context.SecurityContextPersistenceFilter@de18f63, org.springframework.security.web.header.HeaderWriterFilter@5c0f79f0, org.springframework.security.web.csrf.CsrfFilter@6ae7deac, org.springframework.security.web.authentication.logout.LogoutFilter@7bbbb6a8, org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter@4a68135e, org.springframework.security.web.authentication.www.BasicAuthenticationFilter@50a3d0f6, eu.openanalytics.containerproxy.util.AppRecoveryFilter@4b1abd11, org.springframework.security.web.savedrequest.RequestCacheAwareFilter@6345e7b9, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@b18c4, org.springframework.security.web.authentication.AnonymousAuthenticationFilter@6e57b5e9, org.springframework.security.web.session.SessionManagementFilter@3daa82be, org.springframework.security.web.access.ExceptionTranslationFilter@365553de, eu.openanalytics.shinyproxy.AuthenticationRequiredFilter@1abc9f14, org.springframework.security.web.access.intercept.FilterSecurityInterceptor@66f659e6]
2023-04-27 14:08:46.004  INFO 1 --- [           main] o.s.b.a.w.s.WelcomePageHandlerMapping    : Adding welcome page template: index
2023-04-27 14:08:49.169  INFO 1 --- [           main] o.s.l.c.support.AbstractContextSource    : Property 'userDn' not set - anonymous context will be used for read-write operations
2023-04-27 14:08:49.780  INFO 1 --- [           main] o.s.i.endpoint.EventDrivenConsumer       : Adding {logging-channel-adapter:_org.springframework.integration.errorLogger} as a subscriber to the 'errorChannel' channel
2023-04-27 14:08:49.781  INFO 1 --- [           main] o.s.i.channel.PublishSubscribeChannel    : Channel 'ContainerProxy.errorChannel' has 1 subscriber(s).
2023-04-27 14:08:49.784  INFO 1 --- [           main] o.s.i.endpoint.EventDrivenConsumer       : started bean '_org.springframework.integration.errorLogger'
2023-04-27 14:08:49.801  INFO 1 --- [           main] io.undertow                              : starting server: Undertow - 2.2.21.Final
2023-04-27 14:08:49.816  INFO 1 --- [           main] org.xnio                                 : XNIO version 3.8.8.Final
2023-04-27 14:08:49.832  INFO 1 --- [           main] org.xnio.nio                             : XNIO NIO Implementation Version 3.8.8.Final
2023-04-27 14:08:50.130  INFO 1 --- [           main] org.jboss.threads                        : JBoss Threads version 3.1.0.Final
2023-04-27 14:08:50.270  INFO 1 --- [           main] o.s.b.w.e.undertow.UndertowWebServer     : Undertow started on port(s) 8080 (http)
2023-04-27 14:08:50.515  INFO 1 --- [           main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2023-04-27 14:08:50.515  INFO 1 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 227 ms
2023-04-27 14:08:50.535  INFO 1 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2023-04-27 14:08:51.901  INFO 1 --- [           main] io.undertow                              : starting server: Undertow - 2.2.21.Final
2023-04-27 14:08:51.908  INFO 1 --- [           main] o.s.b.w.e.undertow.UndertowWebServer     : Undertow started on port(s) 9090 (http)
2023-04-27 14:08:52.002  INFO 1 --- [           main] e.o.c.service.AppRecoveryService         : Recovery of running apps disabled
2023-04-27 14:08:52.003  INFO 1 --- [           main] e.o.c.util.StartupEventListener          : Started ShinyProxy 3.0.1 (ContainerProxy 1.0.1)
2023-04-27 14:08:57.427  INFO 1 --- [  XNIO-1 task-1] io.undertow.servlet                      : Initializing Spring DispatcherServlet 'dispatcherServlet'
2023-04-27 14:08:57.428  INFO 1 --- [  XNIO-1 task-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2023-04-27 14:08:57.434  INFO 1 --- [  XNIO-1 task-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2023-04-27 14:09:05.338  INFO 1 --- [  XNIO-1 task-5] e.o.containerproxy.service.UserService   : User logged in [user: jack]
2023-04-27 14:09:07.534  INFO 1 --- [pool-2-thread-1] e.o.containerproxy.service.ProxyService  : [user=jack proxyId=02deb45e-9d18-4516-a512-0e0518fef70f specId=rstudio] Starting proxy
2023-04-27 14:09:14.957  INFO 1 --- [pool-2-thread-1] e.o.containerproxy.service.ProxyService  : [user=jack proxyId=02deb45e-9d18-4516-a512-0e0518fef70f specId=rstudio] Proxy activated

```

**The interessting part is:** 

In the RStudio Container i find the file under:  /home/rstudio/.local/share/rstudio/uploaded-files/dad7-4aed-c449-ab27.bin

Any help are is appreciated.

**Comment by JoseMMerida** [comment]
I have the same issue

**Comment by LEDfan** [comment]
Hi

Thank you for reporting this. We just fixed this and created a snapshot release so that you can use/test the fix.

You can try this build using the docker image `openanalytics/shinyproxy-snapshot:3.0.2-SNAPSHOT-20230515.131226` or using the following jar: https://nexus.openanalytics.eu/repository/snapshots/eu/openanalytics/shinyproxy/3.0.2-SNAPSHOT/shinyproxy-3.0.2-20230515.131226-3.jar

**Comment by 0Xilef** [comment]
Hello,

I have tested the snapshot and it works.
Upload and download without problems.

Big Thanks and keep up the good work 😁

**Comment by LEDfan** [comment]
FYI: This is now included in the official 3.0.2 release.
